File: src/main/java/sfuosdev/com/emailbackend/Types/TeamType.java
package sfuosdev.com.emailbackend.Types;

public enum TeamType {
    EVENT,
    TECHNOLOGY,
    STRATEGY;

    public static TeamType getTeamType(String team) {
        return TeamType.valueOf(team.toUpperCase());
    }

    public static String getTeamTypeString(TeamType teamType) {
        return teamType.toString();
    }
}

File: enums/team_type.py
from enum import Enum


class TeamType(str, Enum):
    EVENT = "EVENT"
    TECHNOLOGY = "TECHNOLOGY"
    STRATEGY = "STRATEGY"

    @classmethod
    def from_string(cls, team: str) -> "TeamType":
        try:
            return cls[team.upper()]
        except KeyError:
            raise ValueError(
                f"Invalid team type: '{team}'. Must be one of: {[e.value for e in cls]}"
            )
